File: public/public-web/src/intTestDisabled/java/ch/difty/scipamato/publ/config/ScipamatoPropertiesIntegrationTest.java
package ch.difty.scipamato.publ.config;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import ch.difty.scipamato.publ.ScipamatoPublicApplication;

@SpringBootTest(classes = ScipamatoPublicApplication.class)
class ScipamatoPropertiesIntegrationTest {

    @Autowired
    private ScipamatoProperties sp;

    @Test
    void brand_hasDefaultValue() {
        assertThat(sp.getBrand()).isEqualTo("SciPaMaTo");
    }

    @Test
    void defaultLocalization_hasDefaultEnglish() {
        assertThat(sp.getDefaultLocalization()).isEqualTo("de");
    }

    @Test
    void pubmedBaseUrl_hasDefaultValue() {
        assertThat(sp.getPubmedBaseUrl()).isEqualTo("https://www.ncbi.nlm.nih.gov/pubmed/");
    }

    @Test
    void responsiveIFrameSupport_isDisabled() {
        assertThat(sp.isResponsiveIframeSupportEnabled()).isFalse();
    }

    @Test
    void multiSelectBoxActionBoxWithMoreEntriesThan_hasDefault4() {
        assertThat(sp.getMultiSelectBoxActionBoxWithMoreEntriesThan()).isEqualTo(4);
    }
}

File: public/public-web/src/integration-test/kotlin/ch/difty/scipamato/publ/config/ScipamatoPropertiesIntegrationTest.kt
package ch.difty.scipamato.publ.config

import ch.difty.scipamato.publ.AbstractIntegrationTest
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeFalse
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired

internal class ScipamatoPropertiesIntegrationTest : AbstractIntegrationTest() {

    @Autowired
    private lateinit var sp: ScipamatoProperties

    @Test
    fun brand_hasDefaultValue() {
        sp.brand shouldBeEqualTo "SciPaMaTo"
    }

    @Test
    fun defaultLocalization_hasDefaultEnglish() {
        sp.defaultLocalization shouldBeEqualTo "de"
    }

    @Test
    fun pubmedBaseUrl_hasDefaultValue() {
        sp.pubmedBaseUrl shouldBeEqualTo "https://www.ncbi.nlm.nih.gov/pubmed/"
    }

    @Test
    fun responsiveIFrameSupport_isDisabled() {
        sp.responsiveIframeSupportEnabled.shouldBeFalse()
    }

    @Test
    fun multiSelectBoxActionBoxWithMoreEntriesThan_hasDefault4() {
        sp.multiSelectBoxActionBoxWithMoreEntriesThan shouldBeEqualTo 4
    }
}
